loom/script/native/core/system/lmString.cpp

```diff
@@ -373,6 +373,15 @@ class LSString {
             endIndex = svalueLength - (-(endIndex + 1));
         }
 
+        if (startIndex < 0)
```

Comment:
This is to sync slice() up with Actionscript functionality so that if a -ve startIndex is provided, it means we want to slice that many characters from the *end* of the string, not the start.  From Adobe docs:

start: Number - The zero-based index of the starting point for the slice. If start is a negative number, the starting point is determined from the end of the string, where -1 is the last character.

I've added a comment to the code to be clear :)